src/acat/azext_acat/custom.py

```diff
@@ -0,0 +1,537 @@
+        args_schema.events._required = False
+        args_schema.events._registered = False
+        args_schema.payload_url._required = False
+        args_schema.payload_url._registered = False
+        args_schema.status._required = False
+        args_schema.status._registered = False
+        args_schema.update_webhook_key._required = False
+        args_schema.update_webhook_key._registered = False
+
+        args_schema.trigger_mode = AAZStrArg(
+            options=["--trigger-mode"],
+            arg_group="Properties",
+            help="whether to send notification under any event.",
+            enum={"all": "true", "customize": "false"},
+        )
+        args_schema.events_with_default = AAZListArg(
+            options=["--events"],
+            arg_group="Properties",
+            help="under which event notification should be sent.",
+        )
+        args_schema.events_with_default.Element = AAZStrArg(
+            enum={
+                "assessment_failure": "assessment_failure",
+                "generate_snapshot_failed": "generate_snapshot_failed",
+                "generate_snapshot_success": "generate_snapshot_success",
+                "report_configuration_changes": "report_configuration_changes",
+                "report_deletion": "report_deletion",
+            },
+        )
+        args_schema.status_nullable = AAZStrArg(
+            options=["--disable"],
+            arg_group="Properties",
+            help="Webhook status.",
+            nullable=True,
+            enum={"false": "Enabled", "true": "Disabled"},
+            # default="enalbe",
+            # blank="disable"
+        )
+        args_schema.payload_url_nullable = AAZStrArg(
+            options=["--payload-url"],
+            arg_group="Properties",
+            help="webhook payload url",
+            nullable=True,
+            fmt=AAZStrArgFormat(
+                pattern=r"^(http(s)?://)[\S]{0,64994}$",
+            ),
+        )
+        args_schema.webhook_key_with_default = AAZStrArg(
+            options=["--secret"],
+            arg_group="Properties",
+            help="webhook secret token. If not set, this field value is null; otherwise, please set a string value.",
+            nullable=True,
+            fmt=AAZStrArgFormat(
+                pattern="^.{0,2048}$",
+            ),
+        )
+        return args_schema
+
+    def pre_operations(self):
+        from azure.cli.core.aaz.utils import assign_aaz_list_arg
+
+        args = self.ctx.args
+        args.events = assign_aaz_list_arg(args.events, args.events_with_default)
+        args.payload_url = args.payload_url_nullable
+        args.webhook_key = args.webhook_key_with_default
+        args.status = args.status_nullable
+        args.send_all_events = args.trigger_mode
+
+        wKey = str(args.webhook_key)
+        hasValidWebhookKey = wKey != "Undefined" and len(wKey) > 0
+        args.update_webhook_key = "true" if hasValidWebhookKey else "false"
```

Comment:
Same here. And when the user set `webhook_key` as null, the service support to remove the key right? If that's true, then `update_webhook_key` should be true right? 
```suggestion
        If has_value(args.webhook_key):
            args.update_webhook_key = "true"
        else:
            args.update_webhook_key = "false"
```